src/analysis/normed_space/deriv.lean

```diff
@@ -0,0 +1,198 @@
+/-
+Copyright (c) 2019 Jeremy Avigad. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Jeremy Avigad
+
+The Fréchet derivative.
+
+Let `E` and `F` be normed spaces, and `f : E → F`. Then
+
+  `has_fderiv_at_within f f' x s`
+
+says that the function `f' : E → F` is a bounded linear map and `f` has derivative `f'` at
+`x`, where the domain of interest is restricted to `s`. We also have
+
+  `has_fderiv_at f f' x := has_fderiv_at_within f f' x univ`
+
+The derivative is defined in terms of the `littleo` relation, but also characterized in terms of
+the `tendsto` relation.
+-/
+import topology.basic analysis.normed_space.bounded_linear_maps ..asymptotics tactic.abel
+open filter asymptotics
+
+variables {E : Type*} [normed_space ℝ E]
```

Comment:
Is there really something specific in this file to real numbers, or would everything go through for the complex numbers or the p-adics, say?

Reply:
I just tried changing the variable declarations to this:
```lean
variables {K : Type*} [normed_field K]
variables {E : Type} [normed_space K E]
variables {F : Type} [normed_space K F]
variables {G : Type} [normed_space K G]

include K
```
I don't see why it shouldn't work, since ultimately norms are put on everything. But the declaration causes the elaborator to fail miserably everywhere. Tracing class instances doesn't make it clear what is going wrong. One guess is that the problem has to do with the fact that a `normed_field` is also an instance of a `normed_space` (I think).

I move to accept the PR as is, so that anyone who wants to get the generalization to work can give it a try.

Reply:
I think it fails because `normed_space` etc are no longer `out_param`, so you have to make `K` explicit everywhere.